contracts/src/LightClientV2.sol

```diff
@@ -24,17 +29,19 @@ contract LightClientV2 is LightClient {
     error DeprecatedApi();
 
     /// @notice Initialize V2
-    function initializeV2(uint64 blocksPerEpoch) public reinitializer(2) {
-        _initializeV2(blocksPerEpoch);
-    }
-
-    /// @dev Avoid initialization problem during testing, avoid always upgrading from V1
-    function _initializeV2(uint64 blocksPerEpoch) internal {
-        if (blocksPerEpoch == 0) {
+    function initializeV2(uint64 _blocksPerEpoch, uint64 _epochStartBlock)
```

Comment:
So the plan is to deploy the LC update after nodes have done the PoS Upgrade? Or otherwise how do we know the block at which epochs are enabled?

Reply:
it's scheduled, i assume it's part of the `HotShotConfig`, set ahead of time. Also only `firstEpoch` matters, the exact `epochStartBlock` doesn't matter that much. as @ss-es pointed out, it's soft promise, that it will kickstart in that epoch.